Plugins/BridgeJS/Sources/BridgeJSSkeleton/BridgeJSSkeleton.swift

```diff
@@ -70,17 +70,40 @@ public struct ABINameGenerator {
 /// Context for bridge operations that determines which types are valid
 public enum BridgeContext: Sendable {
     case importTS
-    case protocolExport
+    case exportSwift
 }
 
-public enum BridgeType: Codable, Equatable, Sendable {
+public struct ClosureSignature: Codable, Equatable, Hashable, Sendable {
+    public let parameters: [BridgeType]
+    public let returnType: BridgeType
+    public let isAsync: Bool
+    public let isThrows: Bool
+    public let mangleName: String
+
+    public init(parameters: [BridgeType], returnType: BridgeType, isAsync: Bool = false, isThrows: Bool = false) {
+        self.parameters = parameters
+        self.returnType = returnType
+        self.isAsync = isAsync
+        self.isThrows = isThrows
+
+        let paramPart =
+            parameters.isEmpty
+            ? "Void"
+            : parameters.map { $0.mangleTypeName }.joined(separator: "_")
+        let returnPart = returnType.mangleTypeName
+        self.mangleName = "\(paramPart)_To_\(returnPart)"
```

Comment:
Can we encode the module name into the mangled name to avoid conflicts with the same name types from other modules?